geps/gep-2907/index.md

```diff
@@ -0,0 +1,109 @@
+## Goals
+* Define high-level terminology for how we refer to TLS in Gateway API.
+* Define top level fields where TLS configuration can live.
+
+## Non-Goals
+* Add or change fields directly. (This may inspire changes in other GEPs
+  though).
+* Commit to including specific parts of TLS configuration in Gateway API. (This
+  is merely to provide space for future configuration, not a commitment that we
+  will add it to the API.)
+
+## Introduction
+
+There are essentially 4 different "buckets" that could be relevant to TLS
+configuration in Gateway API:
+
+```mermaid
+flowchart LR
+    Frontend -.->|1. Client Cert| Gateway
+    Gateway -.->|2. Server Cert| Frontend
+    Gateway -.->|3. Client Cert| Backend
+    Backend -.->|4. Server Cert| Gateway
+```
+
+### Where it can fit
+We have three different places where this configuration could live:
+
+#### A. Gateways
+We can add config that applies to an entire Gateway. Config attached at this
+layer will apply to everything behind a single VIP.
+
+#### B. Gateway Listeners
+Each Gateway contains multiple “Listeners”. Each Listener in a Gateway must have
+a unique combination of protocol, port, and hostname (SNI in the case of HTTPS).
+TLS configuration attached at this layer needs to be isolated via SNI.
+
+#### C. BackendTLSPolicy
+This policy allows us to attach unique TLS configuration per Backend. Depending
+on the organization, this policy may be owned by the application owner or the LB
+admin. Note that this configuration will be used by all Gateways (potentially
+from different implementations) that are connecting to the backend.
+
+## Proposed Buckets
+Note that this does not represent any form of commitment that any of these
+fields or concepts will exist within Gateway API. If they did, they would follow
+the following naming structure though:
+
+### 1. Validate Client Certificate provided by Frontend
+
+| Proposed Placement | Name | Status |
+|-|-|-|
+| Gateway Listener | `Listener.TLS.FrontendValidation` | Proposed |
+
+#### Rationale
+Use FrontendValidation to leave room for concepts like trust anchors and trust
+domains. Anything not strictly tied to validation would belong to Listener.TLS
+which is now reserved exclusively for “Frontend” TLS configuration (1+2).
+
+### 2. Configure TLS Termination, including Server Certificate
+
+| Proposed Placement | Name | Status |
+|-|-|-|
+| Gateway Listener | `Listener.TLS` | GA |
+
+#### Rationale
+This is already finalized in the API and so we're stuck with this name. In
+hindsight a name that was more clearly tied to frontend TLS would have been
+ideal here.
+
+
+### 3. Configure Client Certificate that Gateway should use to connect to Backend
+
+| Proposed Placement | Name | Status |
+|-|-|-|
+| Gateway | `Gateway.BackendTLS.ClientCertificateRef` | Not Proposed |
+| BackendTLSPolicy | `BackendTLSPolicy.ClientCertificateRef` | Not Proposed |
+
+#### Rationale
+It's not yet obvious which of the above options are preferable, but a case could
```

Comment:
Could you add details on why Listener.TLS.BackendValidation wouldn't work?  Wouldn't it be analogous to Listener.TLS.FrontendValidation that was proposed?

Reply:
Added a paragraph to cover this. In general, having FrontendValidation at the Listener level allows a Gateway to accept different client certs per SNI (a matching attribute of the frontend -> Gateway connection). On the other hand, when determining the identity a Gateway should use when connecting to a backend, it should likely either be tied directly to the Gateway or Backend, but the Listener is not particularly relevant in this context.